octobot/updater/binary_updater.py

```diff
@@ -0,0 +1,144 @@
+            self.logger.debug(f"Error when parsing latest binary version : {e}")
+        return None
+
+    async def _get_asset_from_release_data(self):
+        latest_release_data = await self._get_latest_release_data()
+        release_asset_name = self._create_release_asset_name(os_util.get_os())
+        return release_asset_name, self._get_asset_from_name(latest_release_data, release_asset_name)
+
+    def _get_asset_from_name(self, release_data, expected_asset_name):
+        try:
+            for asset in release_data["assets"]:
+                asset_name, _ = os.path.splitext(asset["name"])
+                if expected_asset_name == asset_name:
+                    return asset
+        except KeyError as e:
+            self.logger.debug(f"Error when searching for {expected_asset_name} in latest release data : {e}")
+        return None
+
+    def _create_release_asset_name(self, platform):
+        if platform is commons_enums.PlatformsName.WINDOWS:
+            if os_util.is_machine_64bit():
+                return f"{constants.PROJECT_NAME}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryPlatformsName.WINDOWS.value}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryArchitectureName.X64_X86.value}{DeliveryPlatformsExtension.WINDOWS.value}"
+        elif platform is commons_enums.PlatformsName.MAC:
+            if os_util.is_machine_64bit():
+                if os_util.is_machine_64bit():
+                    return f"{constants.PROJECT_NAME}{self.BINARY_DELIVERY_SEPARATOR}" \
+                           f"{DeliveryPlatformsName.MAC.value}{self.BINARY_DELIVERY_SEPARATOR}" \
+                           f"{DeliveryArchitectureName.X64_X86.value}{DeliveryPlatformsExtension.MAC.value}"
+        elif platform is commons_enums.PlatformsName.LINUX:
+            if os_util.is_machine_64bit():
+                return f"{constants.PROJECT_NAME}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryPlatformsName.LINUX.value}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryArchitectureName.X64_X86.value}{DeliveryPlatformsExtension.LINUX.value}"
+            elif os_util.is_arm_machine():
+                return f"{constants.PROJECT_NAME}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryPlatformsName.LINUX.value}{self.BINARY_DELIVERY_SEPARATOR}" \
+                       f"{DeliveryArchitectureName.ARM_X64.value}{DeliveryPlatformsExtension.LINUX.value}"
+        return None
+
+    async def _download_binary(self):
+        release_asset_name, matching_asset_binary = await self._get_asset_from_release_data()
+        new_binary_file = f"{release_asset_name}{self.NEW_BINARY_SUFFIX}"
+        if matching_asset_binary is None:
+            self.logger.error(f"Error when downloading latest version binary : Release not found on server")
+            return None
+        async with aiofiles.open(new_binary_file, 'wb+') as downloaded_file:
```

Comment:
I think we should log something before starting the download to tell what is going on